packages/mcp-auth/src/index.ts

```diff
@@ -210,19 +281,68 @@ export class MCPAuth {
 
       switch (modeOrVerify) {
         case 'jwt': {
-          if (!jwksUri) {
-            throw new MCPAuthAuthServerError('missing_jwks_uri');
-          }
+          return async (token: string) => {
+            const unverifiedJwtPayload = decodeJwt(token);
+
+            if (!unverifiedJwtPayload.iss) {
+              throw new MCPAuthBearerAuthError('invalid_token', {
+                cause: 'The JWT payload does not contain the `iss` field or it is malformed.',
+              });
+            }
+
+            const authServer = this.getAuthServerMetadataByIssuer(unverifiedJwtPayload.iss);
 
-          return createVerifyJwt(createRemoteJWKSet(new URL(jwksUri), remoteJwkSet), jwtVerify);
+            if (!authServer) {
+              throw new MCPAuthBearerAuthError('invalid_issuer', {
+                expected: this.availableAuthServers
+                  .map((authServer) => authServer.metadata.issuer)
+                  .join(', '),
+                actual: unverifiedJwtPayload.iss,
+              });
+            }
```

Comment:
this looks duplicate to `this.validateIssuer`?

Reply:
Yeah, updated.